lib/board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def cells
+    @cells
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def full?
+    @cells.each do |token|
+      if token == " "
+        return false
+      end
+    end
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |space|
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```